packages/bsky/src/hydration/hydrator.ts

```diff
@@ -724,3 +755,23 @@ const mergeManyStates = (...states: HydrationState[]) => {
 const mergeManyMaps = <T>(...maps: HydrationMap<T>[]) => {
   return maps.reduce(mergeMaps, undefined as HydrationMap<T> | undefined)
 }
+
+const actionTakedownLabels = <T>(
+  keys: string[],
+  hydrationMap: HydrationMap<T>,
+  labels: Labels,
+) => {
+  for (const key of keys) {
+    const subjectLabels = labels.get(key)
+    if (!subjectLabels) continue
+    if (includesTakedownLabel(subjectLabels)) {
+      hydrationMap.set(key, null)
+    }
+  }
+}
+
+const includesTakedownLabel = (labels: Label[]) => {
+  return labels.some((l) => l.val === TAKEDOWN_LABEL)
```

Comment:
Is it guaranteed that none of the labels here have a `neg: true`?

Reply:
Should be, but I'll add a check just in case 👍